src/engine/engine_wt.c

```diff
@@ -117,16 +112,20 @@ static void _ocf_write_wt_req_complete(struct ocf_request *req)
 				OCF_QUEUE_ALLOW_SYNC | OCF_QUEUE_PRIO_HIGH);
 	} else {
 		ocf_req_unlock_wr(ocf_cache_line_concurrency(req->cache), req);
-
-		req->complete(req, req->info.core_error ? req->error : 0);
-
+		req->complete(req, 0);
 		ocf_req_put(req);
 	}
 }
 
 static void _ocf_write_wt_cache_complete(struct ocf_request *req, int error)
 {
 	if (error) {
+		/* Cache error code is not propagated further to the user here
+		 * because data could be successfully written to the core device
+		 * despite the cache IO error.
+		 * Error flag is set though to indicate that the error occurred
+		 * and to invalidate the request in completion. */
```

Comment:
If we don't want to propagate the cache error code to the user, is line 129 required? I understand this ensures `req->error` is always != 0 and the check in line 132 is obsolete.

Reply:
Fixed.